### ISSUE TITLE
Crash in research_finish_item

### ISSUE THREAD

**Issue by janisozaur** [OpenRCT2/OpenRCT2#3780]
**OS:** Windows
**Commit:** fbcc68d

Taken from https://openrct2.org/forums/topic/1330-crashing-problem-with-scenarios-downloaded-from-nedesigns/
Post too long to paste it here.
TL;DR: "Universal's Hollywood Hills" (2 parks in 1 DL) and "Meersendael" from NEdesigns crashes when changing object selection

**Steps to reproduce:**
1. Open attached save
2. Open `object selection`
3. Select some item
4. Close `object selection`
5. Crash

**Save game:**
Save game can reproduce the crash
[cadfef15-b198-4c39-bde1-1a35892d4bc3.sv6.txt](https://github.com/OpenRCT2/OpenRCT2/files/290119/cadfef15-b198-4c39-bde1-1a35892d4bc3.sv6.txt)

**Backtrace:**
```
>>> p rideEntry
$1 = (rct_ride_entry *) 0xffffffff
>>> bt
#0  0x01501a0f in research_finish_item (entryIndex=71239) at /home/janisozaur/workspace/OpenRCT2/src/management/research.c:194
#1  0x01501f28 in sub_684AC3 () at /home/janisozaur/workspace/OpenRCT2/src/management/research.c:345
#2  0x0146a400 in window_editor_object_selection_close (w=0x178b360 <g_window_list+1216>) at /home/janisozaur/workspace/OpenRCT2/src/windows/editor_object_selection.c:780
#3  0x0149af00 in window_event_close_call (w=0x178b360 <g_window_list+1216>) at /home/janisozaur/workspace/OpenRCT2/src/interface/window.c:1813
#4  0x01498ac6 in window_close (window=0x178b360 <g_window_list+1216>) at /home/janisozaur/workspace/OpenRCT2/src/interface/window.c:661
#5  0x0146a472 in window_editor_object_selection_mouseup (w=0x178b360 <g_window_list+1216>, widgetIndex=2) at /home/janisozaur/workspace/OpenRCT2/src/windows/editor_object_selection.c:802
#6  0x0149af2b in window_event_mouse_up_call (w=0x178b360 <g_window_list+1216>, widgetIndex=2) at /home/janisozaur/workspace/OpenRCT2/src/interface/window.c:1819
#7  0x0154f543 in input_state_widget_pressed (x=956, y=281, state=2, widgetIndex=2, w=0x178b360 <g_window_list+1216>, widget=0x16f91c0 <window_editor_object_selection_widgets+32>) at /home/janisozaur/workspace/OpenRCT2/src/input.c:1296
#8  0x0154d31a in game_handle_input_mouse (x=956, y=281, state=2) at /home/janisozaur/workspace/OpenRCT2/src/input.c:337
#9  0x0154cd2d in game_handle_input () at /home/janisozaur/workspace/OpenRCT2/src/input.c:151
#10 0x0156903f in game_update () at /home/janisozaur/workspace/OpenRCT2/src/game.c:344
#11 0x01561350 in rct2_update () at /home/janisozaur/workspace/OpenRCT2/src/rct2.c:458
#12 0x01552e7b in openrct2_loop () at /home/janisozaur/workspace/OpenRCT2/src/openrct2.c:392
#13 0x01552cdc in openrct2_launch () at /home/janisozaur/workspace/OpenRCT2/src/openrct2.c:322
#14 0x0150edf2 in main (argc=2, argv=0xffffd374) at /home/janisozaur/workspace/OpenRCT2/src/platform/posix.c:60
```

**Comment by Charlieep** [comment]
I have done a little research and found that these games work in the stable version, but crash upon using the object selector in debugging tools. (89db25e)

**Comment by mrtnptrs** [comment]
The crash still occur using 964cf66 even after #4105

**Comment by IntelOrca** [comment]
No longer crashes, @janisozaur can you confirm?

**Comment by janisozaur** [comment]
yes I can, does not occur anymore.

**Comment by janisozaur** [comment]
This turns out to be duplicate of https://github.com/OpenRCT2/OpenRCT2/issues/3769 and as of 41f026e it still happens (see #4546)
